api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -105,6 +109,85 @@ type ProxyLogging struct {
 	Level map[LogComponent]LogLevel `json:"level,omitempty"`
 }
 
+type ProxyAccessLogging struct {
+	// Format defines the format of access logging.
+	Format ProxyAccessLoggingFormat `json:"format"`
+	// Sink defines the sink of access logging.
+	Sink ProxyAccessLoggingSink `json:"sink"`
+}
+
+type ProxyAccessLoggingFormatType string
+
+const (
+	// ProxyAccessLoggingFormatTypeText defines the text access logging format.
+	ProxyAccessLoggingFormatTypeText ProxyAccessLoggingFormatType = "text"
+	// ProxyAccessLoggingFormatTypeJSON defines the JSON access logging format.
+	ProxyAccessLoggingFormatTypeJSON ProxyAccessLoggingFormatType = "json"
+	// TODO: support format type "mix" in the future.
+)
+
+type ProxyAccessLoggingFormat struct {
+	Type ProxyAccessLoggingFormatType `json:"type,omitempty"`
+	// Text defines the text access logging format, following Envoy access logging formatting,
+	// empty value results in proxy's default access log format.
+	// It's required when the format type is "text".
+	// Envoy [command operators](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#command-operators) may be used in the format.
+	// The [format string documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-strings) provides more information.
+	// +optional
+	Text *string `json:"text,omitempty"`
+	// Fields is additional attributes that describe the specific event occurrence.
+	// Structured format for the envoy access logs. Envoy [command operators](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#command-operators)
+	// can be used as values for fields within the Struct.
+	// It's required when the format type is "json".
+	// +optional
+	Fields *map[string]string `json:"fields,omitempty"`
```

Comment:
the type and name of the field usually match, in this case it doesnt - JSON type and field is called `Fields`

Reply:
You also don't need a pointer to a map because a map is already a reference type

Reply:
yes, it's mistake.